mava/systems/tf/mad4pg/networks.py

```diff
@@ -64,114 +62,20 @@ def make_default_networks(
-        critic_networks_layer_sizes = {
-            key: critic_networks_layer_sizes for key in specs.keys()
-        }
-
-    observation_networks = {}
-    policy_networks = {}
-    critic_networks = {}
-    for key in specs.keys():
-        # TODO (dries): Make specs[key].actions
-        #  return a list of specs for hybrid action space
-        # Get total number of action dimensions from action spec.
-        agent_act_spec = specs[key].actions
-        if type(specs[key].actions) == DiscreteArray:
-            num_actions = agent_act_spec.num_values
-            minimum = [-1.0] * num_actions
-            maximum = [1.0] * num_actions
-            agent_act_spec = BoundedArray(
-                shape=(num_actions,),
-                minimum=minimum,
-                maximum=maximum,
-                dtype="float32",
-                name="actions",
-            )
-
-        # Get total number of action dimensions from action spec.
-        num_dimensions = np.prod(agent_act_spec.shape, dtype=int)
-
-        # An optional network to process observations
-        observation_network = tf2_utils.to_sonnet_module(tf.identity)
-
-        # Create the policy network.
-        if archecture_type == ArchitectureType.feedforward:
-            policy_network = [
-                networks.LayerNormMLP(
-                    policy_networks_layer_sizes[key], activate_final=True
-                ),
-            ]
-        elif archecture_type == ArchitectureType.recurrent:
-            policy_network = [
-                networks.LayerNormMLP(
-                    policy_networks_layer_sizes[key][:-1], activate_final=True
-                ),
-                snt.LSTM(policy_networks_layer_sizes[key][-1]),
-            ]
-
-        policy_network += [
-            networks.NearZeroInitializedLinear(num_dimensions),
-            networks.TanhToSpec(agent_act_spec),
-        ]
-
-        # Add Gaussian noise for simple exploration.
-        if sigma and sigma > 0.0:
-            policy_network += [
-                networks.ClippedGaussian(sigma),
-                networks.ClipToSpec(agent_act_spec),
-            ]
-
-        policy_network = policy_network_func(policy_network)
-
-        # Create the critic network.
-        critic_network = snt.Sequential(
-            [
-                # The multiplexer concatenates the observations/actions.
-                networks.CriticMultiplexer(),
-                networks.LayerNormMLP(
-                    critic_networks_layer_sizes[key], activate_final=True
-                ),
-                DiscreteValuedHead(vmin, vmax, num_atoms),
-            ]
-        )
-        observation_networks[key] = observation_network
-        policy_networks[key] = policy_network
-        critic_networks[key] = critic_network
-
-    return {
-        "policies": policy_networks,
-        "critics": critic_networks,
-        "observations": observation_networks,
-    }
+    return make_default_networks_maddpg(
```

Comment:
Did we merge MADDPG and MAD4PG's networks? It does seem like a great idea as much of the code is shared :+1: Thanks @KaleabTessera :smile:  It might be worth it to just check if both still work as expected.

Reply:
I will run some benchmarks before merging this in :+1: